File: android/src/main/kotlin/com/sabiantech/sabian_native_common/events/AbstractEvent.kt
package com.sabiantech.sabian_native_common.events

import io.flutter.plugin.common.EventChannel

abstract class AbstractEvent : EventChannel.StreamHandler {
    protected var eventSink: EventChannel.EventSink? = null
    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
        eventSink = events
    }

    override fun onCancel(arguments: Any?) {
        eventSink = null
    }

    protected open fun raise(payload: Any) {
        eventSink?.let {
            it.success(payload)
        }
    }
}

File: ios/Classes/events/AbstractEvent.swift
//
//  AbstractEvent.swift
//  Runner
//
//  Created by Sweet Pea on 22/11/2023.
//

import Foundation
import Flutter


class AbstractEvent : NSObject,FlutterStreamHandler {
    
    var eventSink: FlutterEventSink? = nil
    
    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        self.eventSink = events
        return nil
    }
    
   
    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        self.eventSink = nil
        return nil
    }
    

    func raise(_ payload: Any) {
        if let sink = eventSink {
            sink(payload)
        }
    }
}
